src/Facets/HopFacetOptimized.sol

```diff
@@ -0,0 +1,255 @@
+        for (uint256 i; i < bridges.length; i++) {
+            // Give Hop approval to bridge tokens
+            LibAsset.maxApproveERC20(IERC20(tokensToApprove[i]), address(bridges[i]), type(uint256).max);
+        }
+    }
+
+    /// @notice Bridges ERC20 tokens via Hop Protocol from L1
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _hopData data specific to Hop Protocol
+    function startBridgeTokensViaHopL1ERC20(ILiFi.BridgeData calldata _bridgeData, HopData calldata _hopData)
+        external
+        validateBridgeData(_bridgeData)
+    {
+        // Deposit assets
+        LibAsset.depositAsset(_bridgeData.sendingAssetId, _bridgeData.minAmount);
+        // Bridge assets
+        _hopData.hopBridge.sendToL2(
+            _bridgeData.destinationChainId,
+            _bridgeData.receiver,
+            _bridgeData.minAmount,
+            _hopData.destinationAmountOutMin,
+            _hopData.destinationDeadline,
+            address(0),
+            0
+        );
+        emit LiFiTransferStarted(_bridgeData);
+    }
+
+    /// @notice Bridges Native tokens via Hop Protocol from L1
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _hopData data specific to Hop Protocol
+    function startBridgeTokensViaHopL1Native(ILiFi.BridgeData calldata _bridgeData, HopData calldata _hopData)
+        external
+        payable
+        validateBridgeData(_bridgeData)
+    {
+        // Deposit assets
+        LibAsset.depositAsset(_bridgeData.sendingAssetId, _bridgeData.minAmount);
+        // Bridge assets
+        _hopData.hopBridge.sendToL2{ value: _bridgeData.minAmount }(
+            _bridgeData.destinationChainId,
+            _bridgeData.receiver,
+            _bridgeData.minAmount,
+            _hopData.destinationAmountOutMin,
+            _hopData.destinationDeadline,
+            address(0),
+            0
+        );
+        emit LiFiTransferStarted(_bridgeData);
+    }
+
+    /// @notice Performs a swap before bridging ERC20 tokens via Hop Protocol from L1
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _swapData an array of swap related data for performing swaps before bridging
+    /// @param _hopData data specific to Hop Protocol
+    function swapAndStartBridgeTokensViaHopL1ERC20(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        HopData calldata _hopData
+    ) external validateBridgeData(_bridgeData) {
+        // Deposit and swap assets
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+
+        // Bridge assets
+        uint256 value = LibAsset.isNativeAsset(address(_bridgeData.sendingAssetId)) ? _bridgeData.minAmount : 0;
```

Comment:
This is needed in the case that we do a swap that starts with an ERC20 but ends up bridging via native.

Reply:
Well, then how about we also split this function like we did with the others? 
Will save us another check, therefore even better gas optimization?

Reply:
I am not too sure about those swapAndBridge functions, currently this one can be triggered with ERC20 and can also bridge either ERC20 or native. The other function could also be called with both but can only bridge native. I think it would make sense to either keep it flexible and use only one function or specify in the signature if we will trigger hop with native or a ERC20.

Reply:
Yeah thinking about this now we wouldn't actually call the ERC20 method in the case native is bridged. I originally did this to make the tests work but we don't need this.

Reply:
Fixed